doc/develop/optimizations/tools.rst

```diff
@@ -12,9 +15,29 @@ The build system offers 3 targets to view and analyse RAM, ROM and stack usage
 in generated images. The tools run on the final image and give information
 about size of symbols and code being used in both RAM and ROM. Additionally,
 with features available through the compiler, we can also generate worst-case
-stack usage analysis:
+stack usage analysis.
+
+Hidden
+  The RAM and ROM reports list all processing symbols with no matching mapped files
+  in the Hidden category.
+
+  This means that the file for the listed symbol was not added to the metadata file,
+  was empty or was undefined. The tool was unable to get the name
+  of the function for the given symbol nor identify where it comes from.
+
+No paths
+  The RAM and ROM reports list all processing symbols with relative paths
+  in the No paths category.
+
+  This means that the listed symbols cannot be placed in the tree structure
+  of the report at an absolute path under one specific file. The tool was able
+  to get the name of the function, but it was unable to identify where it comes from.
+
+  .. note::
+
+     You can have multiple cases of the same function, and the No paths category
+     will list the sum of these in one entry.
```

Comment:
This doesn't really flow well with the previous paragraph, and would need some context for the reader who likely hasn't even used the tools just yet and therefore has no idea what "Hidden" or "No paths" refers to.

Maybe starting with something like "Some of the tools mentioned in this section are organizing their output based on the physical organization of the symbols. As some symbols might be external to the project's tree structure, or might lack metadata needed to display them by name, the following top-level containers are used to group such symbols::" would help? And maybe this whole paragraphe would benefit from being a `.. note::` (if yes, the inner note ("You can have multiple...") would probably work better visually as a simple sentence)

Reply:
Good idea, @kartben . I've made the edits, but with the liberty of putting the descriptions of Hidden and No paths into a table.